### DISCUSSION TITLE
Motivation and direction for storage market development

### DISCUSSION THREAD

**Discussion by anorth** [filecoin-project/FIPs#241] [General]
Over the first year or so since Filecoin network liftoff, greater-than-expected storage provider onboarding forced us to focus a lot on network scaling and other issues associated with being a young network. Now that we seem to have that under control (and with [proposals](https://github.com/filecoin-project/FIPs/discussions/240) in our pocket in case growth explodes again) I expect we will spend a lot more focus on the functionality of Filecoin as a storage marketplace. But not only from a core consensus protocol point of view! The Filecoin VM promises to unlock the creative potential of thousands of developers to build applications on and in Filecoin. Thus, I hope and expect that a large majority of the development of storage market features and functionality to come from non-builtin smart contracts developed by teams outside the Filecoin protocol core. I'm posting this to provide context for the changes I'll be proposing and developing in the near future.

Features like deal extension, multi-sector deals, re-negotiation, deal transfer, capacity deals, auction markets, repair markets, insurance, derivatives etc should all be possible in smart contracts. None should require a network-wide upgrade to realise. I hope for a variety of marketplace mechanics, bound by some shared standards and interfaces, just as there are variety in token exchanges and marketplaces on other blockchains.

However, even if we had the FVM today, much of the above would be impossible to build. This is because the structure of the built-in storage miner and storage markets does not provide the information and hooks that would be needed by these applications, and privileges the built-in market actor as the one which miner actors consult. In order to enable the development of actually-useful storage-related applications and features, we need to break the tight coupling between these built-in actors, expose composable on-chain primitives, and make the capabilities of the built-in market actor available to user-programmable contracts.

Breaking this coupling will also have other benefits. It will significantly speed up future development of both the storage power accounting and future deal markets, support much more scalable data representations, remove the storage market from consensus-critical risk surface, and move subsequent market development into the realm of smart-contract upgrades, rather than network-wide consensus protocol upgrades. This direction is thus motivated by speed, scale, flexibility and programmability.

I expect this direction to be the main focus of my work over the coming months. There are a number of significant changes needed to establish this open playing field. My goal is that, by the time the FVM enables user-programmable contracts, the stage is set for an explosion in application and feature innovation to match the stupendous uptake of storage-powered consensus by providers in the network's first year.

**Comment by jennijuju** [comment]
This sounds m great!
first question come to mind: do you imagine the current market actor to be the smart contract standard/ interface once the user developed smart contract in unblocked?

How would the new smart contract interact with current market actor, will we be transferring the market  states to new smart contracts, or only allow new smart contracts read the existing state? If it’s the later, how would the newly developed feature via smart contracts being applied or available to the existing storage market and clients ?

**Reply by anorth** [reply]
> do you imagine the current market actor to be the smart contract standard

I think the built-in storage market actor should become a not-builtin actor, demonstrating the level playing field. I do not think its public interface should set the standard for all markets, but I think we should attempt to standardise _parts_ of it. For example, we probably want to represent a deal as something like an NFT, and be transferrable between clients, and expose some standard metadata. We might aim for the smallest "deal token" standard that could be adopted by other markets without restricting their freedom on other fronts.

> How would the new smart contract interact with current market actor

Probably not at all. New markets are their own sovereign, the wouldn't get any internal access to the used-to-be-builtin one. I think it's unlikely that deals would be transferrable between markets initially, but maybe could be worked towards if we could see much benefit.

**Comment by anorth** [comment]
Here's my current understanding of the changes required:
- Make FIL+ subsidy explicit, remove quality-adjusted power
- Move the sector->deal mapping from sector metadata into the market actor
- Expand FIL+ registry to support multiple storage markets, so they can also broker verified deals [edited]

**Reply by Fatman13** [reply]
> FIL+ subsidy, remove quality-adjusted power

Do you mean remove the x10 multiplier and have regular deals only? Or having other ways to subsidize fil+ rather than power multiplier?

**Reply by Fatman13** [reply]
> Extract FIL+ registry from the storage market, so other markets can also broker verified deals

What does it mean by extract FIL+ registry from storage market? Do you mean datacap system only exist within storage market ecosystem right now?

**Reply by dkkapur** [reply]
@anorth would be great to have you join into an upcoming Fil+ Community Governance call and share the proposal with the community! https://github.com/filecoin-project/notary-governance/issues/364 

Else we can discuss as a group and come back here with some comments/thoughts. 

cc @KwrayFF @galen-mcandrew

**Reply by anorth** [reply]
@Fatman13 I mean having other ways to subsidize FIL+ rather than the power multiplier (e.g. an [explicit subsidy](#243)). Note: there may be other ways to achieve sufficient decoupling, but this is my current proposal.

"Extract FIL+ registry" was poor wording: the registry is already implemented separately. But I mean make changes such that an alternative market implementation could also be eligible and able to make FIL+ verified deals.

**Comment by anorth** [comment]
The eagle-eyes may have noticed a divergence from the arguments presented [here](https://github.com/filecoin-project/FIPs/issues/57#issuecomment-877931063) for restricting sector data. 

What's changed is that the FVM will mean that alternative marketplaces can be built _on Filecoin_. The barrier to innovation of preventing the development of alternative on-chain markets is much higher than the barrier of not being able to rely on zero CC data.
On chain markets will be visible, and will pay tx fees in FIL (even if they settle in stablecoin). They are within the Filecoin economy in the same way DEXs and credit platforms are within the Ethereum economy.
